### ISSUE TITLE
ENH spatial.ConvexHull: Adding an option to disable the Qt option to Qhull

### ISSUE THREAD

**Issue by thangckt** [scipy/scipy#10944]
Dear Developers,

I need to count the facets forming the Hull. For example, a simple cubic hull should have 6 facets. But since the option 'Qt' is always enable, the square facets is always triangulated out.

I need to disable option 'Qt', or is there any other way to get 6 for the answer?
Please help
Thank you so much

**Comment by tylerjereddy** [comment]
The triangular simplices that form common facets of the cube can be identified by the matching hyperplane equations (two triangles on the same 2D plane inside 3D space):

```python
import numpy as np
import scipy
from scipy.spatial import ConvexHull

print("scipy.__version__:", scipy.__version__)

# build a cube from Warren's code
# on SO: https://stackoverflow.com/a/52229558/2942522
N = 3 
cube = 2*((np.arange(2**N)[:,None] & (1 << np.arange(N))) > 0) - 1 

hull = ConvexHull(cube)

# there are indeed 12 triangles for 6 square facets:
print("hull.simplices.shape:", hull.simplices.shape)

# if we look at the "equations" of the 2D planes
# (hyperplanes) on which these triangles lie,
# we can immediatley see that consecutive triangles
# have redundant equations (they lie on the same square facet)
print("hull.equations:", hull.equations)

# so, let's just count facets as the unique hyperplane equations
# (square faces):
unique_facet_equations = np.unique(hull.equations, axis=0)
print("unique_facet_equations.shape:", unique_facet_equations.shape)

print("number of facets:", unique_facet_equations.shape[0])

```

Output:

```
scipy.__version__: 1.4.0.dev0+5c6562a
hull.simplices.shape: (12, 3)
hull.equations: [[-0. -0. -1. -1.]
 [-0. -0. -1. -1.]
 [ 0. -1.  0. -1.]
 [ 0. -1.  0. -1.]
 [ 1. -0. -0. -1.]
 [ 1. -0. -0. -1.]
 [-1. -0. -0. -1.]
 [-1. -0. -0. -1.]
 [ 0.  1. -0. -1.]
 [ 0.  1. -0. -1.]
 [-0. -0.  1. -1.]
 [-0. -0.  1. -1.]]
unique_facet_equations.shape: (6, 4)
number of facets: 6

```

**Comment by tylerjereddy** [comment]
> there any other way to get 6 for the answer?

I think I've answered that clearly above with a bit of extra work to get what you need. No additional feedback in the last month, so closing.

**Comment by tylerjereddy** [comment]
We might want to consider documenting the approach in a simple example or something--perhaps open a separate issue for that if this is likely a common requirement.

**Comment by pv** [comment]
duplicate of gh-10442

**Comment by kvroeijenstijn** [comment]
Hello Tyler, 

Thanks for that little script, it helps with post processing the results of the convex hull. I am using the convexHull code in a 3d application where it is important to preserver quad topology where possible. I am currently writing code to merge the triangles back into quads by comparing the planes.

I would be very curious to see if disabling the Qt option would give me what I am expecting out of the box, would you please re-consider allowing this?

Thanks,
Koen

**Comment by davidwbulger** [comment]
I agree with koen-v about this. The above script is a minor hassle, and it becomes a bigger hassle if we have to allow for numerical error rather than simply using numpy.unique.

Since the functionality is already in the qhull library, I would have thought this would be a very easy fix (but maybe there's some non-obvious difficulty).

**Comment by tylerjereddy** [comment]
I'll re-open this for visibility/some kind of consideration, though not sure what the path forward would be given backward compatibility constraints.

**Comment by davidwbulger** [comment]
Thanks Tyler. I don't see how adding a named optional argument "Qt" defaulting to True could break any previous code ... but I may be out of my depth here.

**Comment by tylerjereddy** [comment]
I suppose something like that might be possible. At the very least we'd likely have to parametrize our current tests over the two options to see if there are any current features/attributes, etc. that are problematic with `Qt` disabled.
